frontend2/src/views/Submissions.tsx

```diff
@@ -85,12 +115,139 @@ const Submissions: React.FC = () => {
+
+    const fetchNextTourney = async (): Promise<void> => {
+      try {
+        const result = await getNextTournament(episodeId);
+        if (isActiveLookup) {
+          setNextTourneyData(result);
+        }
+      } catch (err) {
+        if (isActiveLookup) {
+          setNextTourneyData(undefined);
+        }
+        console.error(err);
+      } finally {
+        if (isActiveLookup) {
+          setNextTourneyLoading(false);
+        }
+      }
+    };
+
+    void fetchNextTourney();
+
+    return () => {
+      isActiveLookup = false;
+    };
+  }, [episodeId]);
+
   return (
     <div className="flex h-full w-full flex-col overflow-auto p-6">
       <h1 className="mb-5 text-3xl font-bold leading-7 text-gray-900">
         Submissions
       </h1>
 
+      <SectionCard title="Next Submission Deadline" className="mb-8">
+        {nextTourneyLoading ? (
+          <div className="flex flex-row items-center gap-4">
+            <Spinner size="md" />
+            <span>Loading...</span>
+          </div>
+        ) : (
+          <TournamentCountdown tournament={nextTourneyData} />
+        )}
+      </SectionCard>
+
+      <SectionCard title="Submit Code" className="mb-8">
+        {episode === undefined ? (
+          <>
+            <div className="flex flex-row items-center gap-4">
+              <Spinner size="md" />
+              <span>Loading...</span>
+            </div>
+          </>
+        ) : episode.frozen ? (
+          <>
+            <p>
+              Submissions are currently frozen! This is most likely due to a
+              submission freeze for a tournament. If you think the freeze has
+              passed, try refreshing the page.
+            </p>
+          </>
+        ) : (
+          <>
+            <p>Submit your code using the button below.</p>
+            <p>
+              Create a{" "}
+              <span className="rounded-md bg-gray-200 px-1 py-0.5 font-mono">
+                zip
+              </span>{" "}
+              file of your robot player, and submit it below. The submission
+              format should be a zip file containing a single folder (which is
+              your package name), which should contain{" "}
+              <span className="rounded-md bg-gray-200 px-1 py-0.5 font-mono">
+                RobotPlayer.java
+              </span>{" "}
+              and any other code you have written, for example:
+            </p>
+            <span className="w-full">
+              <p className="rounded-md border border-gray-400 bg-gray-100 px-4 py-3 font-mono">
+                {
+                  "submission.zip --> examplefuncsplayer --> RobotPlayer.java, FooBar.java"
+                }
+              </p>
+            </span>
+            <form
+              // eslint-disable-next-line @typescript-eslint/no-misused-promises
+              onSubmit={(event) => {
+                event.preventDefault();
+                handleSubmit(onSubmit);
+              }}
```

Comment:
Try changing this to 
```suggestion
              onSubmit={handleSubmit(onSubmit)}
```

Reply:
This causes the page to reload... not sure why?